lib/Frontend/ArgsToFrontendOptionsConverter.cpp

```diff
@@ -498,7 +503,66 @@ bool ArgsToFrontendOptionsConverter::setUpImmediateArgs() {
   return false;
 }
 
+bool ArgsToFrontendOptionsConverter::computeModuleAliases() {
+  auto list = Args.getAllArgValues(options::OPT_module_alias);
+  if (!list.empty()) {
+    auto validate = [this](StringRef value, bool allowModuleName) -> bool
+    {
+      if (!allowModuleName) {
+        if (value == Opts.ModuleName ||
+            value == Opts.ModuleABIName ||
+            value == Opts.ModuleLinkName) {
+          Diags.diagnose(SourceLoc(), diag::error_module_alias_forbidden_name, value);
+          return false;
+        }
+      }
+      if (value == STDLIB_NAME) {
+        Diags.diagnose(SourceLoc(), diag::error_module_alias_forbidden_name, value);
+        return false;
+      }
+      if (!Lexer::isIdentifier(value)) {
```

Comment:
Does that mean that emojis are accepted? 🤔

Reply:
Good point :) It's the same checked used by computing module name. Most likely a c99 identifier will be used in practice, but we can add a check to enforce that later if needed.